### ISSUE TITLE
speciesKey predicate

### ISSUE THREAD

**Issue by OrsonDeWitt** [ropensci/rgbif#661]
I think species / speciesKey should be included as possible predicates.
Consider Caliechthrus leucolophus.
In scientificName, it returns two variables: Cacomantis leucolophus (S.Muller, 1840) and Caliechthrus leucolophus (S.Muller, 1840)
Both of them have different taxon keys: 6085624 and 2496386, respectively.
So the only way to download Caliechthrus leucolophus with the synonym is to use "species" or "speciesKey" fields, which are identical for the species, but are not available in rgbif. Sure, for this species using occ_search is possible, but there could be another that has more than 100,000 observations :)

**Comment by jhnwllr** [comment]
@OrsonDeWitt 

I agree that adding `speciesKey` as a predicate key would be useful. 

However, GBIF's default is to download all occurrences from synonyms as well. So the example above would give back all of the occurrences for both names if you used the taxonKey for the **accepted name**. As you can see from the `occ_count()`, the accepted taxonKey (6085624) has slightly more occurrences associated with it than the the synonym (2496386). 


```R 
occ_count(taxonKey=2496386)
```
> 883

```R 
occ_count(taxonKey=6085624)
```
> 803

So using 

```R 
occ_download(pred("taxonKey",2496386))
```

would download everything.

**Comment by dmi3kno** [comment]
I think this can be closed now that #668 has been merged.
